### PR TITLE
Packaging snafu to run from binary

### PR DESCRIPTION
**Pull request by amitsagtani97**
Depends-On: 343

### CODE REVIEW

File: src/snafu/ycsb_wrapper/trigger_ycsb.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+    #!/usr/bin/env python
```
Comment (aakarshg) [inline src/snafu/ycsb_wrapper/trigger_ycsb.py:1]: why the additional spacing? or is it just a typo?

**Comment by aakarshg** [comment]
> Great work @amitsagtani97 !
> Overall looks good, It will need some love on CI tests since there are some points where I believe they will fail due to test file paths handling.
> 
> In the other hand, why are we double nesting the sources under `src/snafu`? Nesting them under snafu/ should be enough and would simplify module imports and improve project readability.
> I've seen we have other projects with this "double-nesting" structure such as touchstone, @aakarshg can you shed some light over this?

Good point, double nesting isn't needed here, maybe just put sources under src/. historically for touchstone when i started it, the idea was to have yaml files located in src, but not in touchstone as its not a module. but as i started working with it, i moved away from yaml files and basically put them as dict inside the source code.

**Comment by amitsagtani97** [comment]
@rsevilla87 , @aakarshg : Removed double nesting and updated the CI test scripts accordingly :+1:

**Comment by amitsagtani97** [comment]
Hi @rsevilla87 , moved the ci tests to root :+1:

File: ci/run_ci.sh [hunk 2]
```diff
@@ -56,7 +56,7 @@ wait_clean
 for dir in ${test_list}; do
   start_time=`date`
   figlet "CI test for ${dir}"
-  if $dir/ci_test.sh; then
+  if src/$dir/ci_test.sh; then
```
Comment (rsevilla87) [inline ci/run_ci.sh:59]: This need to be updated to:
```suggestion
   if $dir/ci_test.sh; then 
```
And -maxdepth needs to set to 2 in the previous `find` statement

Comment (amitsagtani97) [reply ci/run_ci.sh:59]: @rsevilla87  This calls the individual tests in the benchamark directories, which are located in the src/ 
I guess, this needs not to be changed.

Comment (rsevilla87) [reply ci/run_ci.sh:59]: Yeah, but ${dir} already contains the directory structure where ci_test.sh is, which it's populated by :

test_list=`find * -maxdepth 2 -name ci_test.sh -type f -exec dirname {} \;`    <-- Note maxdepth 2
and
test_list=`echo "${diff_list}" | awk -F "/" '{print $1"/"$2}' | uniq`            <-- Note the modified print statement

File: README.md [hunk 3]
```diff
@@ -26,6 +26,18 @@ environments, develop in SNAFU and then write ripsaw benchmark to integrate with
 | Elasticsearch  | Working  |
 | Prom           | Planned  |
 
+
+It is suggested to use a venv to install and run snafu.
+
+```
+python3 -m venv /path/to/new/virtual/environment
```
Comment (aakarshg) [inline README.md:33]: Can you please also add documentation to show snafu can be imported as a python library and run a benchmark wrapper?

**Comment by bengland2** [comment]
@dry923 strange error from fs-drift, it says nothing was stored in ripsaw-fs-drift-results, but Kibana begs to differ!   I looked up same UUID in Kibana and it's all there.   Is there some race condition where it queries ES before the data has been fully ingested?
```
+ echo RUN STATUS DONE
++ indexes='ripsaw-fs-drift-results ripsaw-fs-drift-rsptimes ripsaw-fs-drift-rates-over-time'
++ check_es 5e8e049b-cbbf-552f-8eb7-3f337a591227 'ripsaw-fs-drift-results ripsaw-fs-drift-rsptimes ripsaw-fs-drift-rates-over-time'
++ [[ 2 != 2 ]]
++ uuid=5e8e049b-cbbf-552f-8eb7-3f337a591227
++ index='ripsaw-fs-drift-results ripsaw-fs-drift-rsptimes ripsaw-fs-drift-rates-over-time'
++ for my_index in $index
++ python3 tests/check_es.py -s marquez.perf.lab.eng.rdu2.redhat.com -p 9200 -u 5e8e049b-cbbf-552f-8eb7-3f337a591227 -i ripsaw-fs-drift-results
No result found in ES
++ exit 1
++ cat uuid
cat: uuid: No such file or directory
+ uuid=
```

**Comment by aakarshg** [comment]
@bengland2 @dry923 @rsevilla87 @jtaleric looks like we'll need to [perform a refresh action](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html#indices-refresh) on the index before we search for the results.

**Comment by dry923** [comment]
> @bengland2 @dry923 @rsevilla87 @jtaleric looks like we'll need to [perform a refresh action](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html#indices-refresh) on the index before we search for the results.

I'm all for the refresh. This has definitely bitten us in the CI env a lot. @aakarshg can you point me to how to trigger it from the python script and I'll add it in.

**Comment by rsevilla87** [comment]
> @bengland2 @dry923 @rsevilla87 @jtaleric looks like we'll need to [perform a refresh action](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html#indices-refresh) on the index before we search for the results.

Yeah, I was aware of this. I added a small sleep at https://github.com/cloud-bulldozer/ripsaw/pull/336 . It's not as fancy as I'd like but worked. Thoughts?

**Comment by amitsagtani97** [comment]
Just re-based the dependent ripsaw pr, we can try to rerun tests now.

**Comment by aakarshg** [comment]
can you please rebase agin @amitsagtani97

**Comment by amitsagtani97** [comment]
> can you please rebase agin @amitsagtani97

done :)

**Comment by aakarshg** [comment]
@dry923 and @jtaleric can you please review it when you get a chance? 

I also think this PR should wait for a sign off from either @acalhounRH or @bengland2 ( both is better ;) ), so can you guys please review when you get a chance?

File: README.md (hunk 3 above)
Comment (amitsagtani97) [reply README.md:33]: this pr enables to use it as a binary. Will create a new pr to support using as a library too.

**Comment by rsevilla87** [comment]
Hey @amitsagtani97, can you rebase ripaw's 343, I pushed some changes yesterday to fix the CI issues.

**Comment by amitsagtani97** [comment]
> Hey @amitsagtani97, can you rebase ripaw's 343, I pushed some changes yesterday to fix the CI issues.

Hi, @rsevilla87 done :+1:

**Pull request merged**